hack/conformance/check_conformance_test_requirements.go

```diff
@@ -30,7 +30,7 @@ import (
 )
 
 const (
-	//e.g. framework.ConformanceIt("should provide secure master service ", func() {
+	//e.g. framework.ConformanceIt("should provide secure master service ", func(ctx context.Context) {
```

Comment:
why don't you add a `hack/verify-ginkgo-context.sh`and `hack-update-ginkgo-context.sh` so we are consistent with this ?
It also will give us more confident in this change having the scripts in repo

Reply:
Not all callbacks need a context, only those that call functions which need a context. In practice, almost all tests will need it because they communicate with a cluster, which implies that they could block and thus should use a context. But should we make it a requirement for all tests?

There may be some value in it. Developers might be tempted to write a test like this because they don't know better:
```
ginkgo.It("...", func() {
    ctx := context.TODO() // For framework helper functions below.
   ...
    e2epod.WaitForPod(ctx, ...)
})
```

The sed invocation isn't perfect, though. A few tests were defined with
```
ginkgo.It("...",
   func() {
   }
)
```

Because sed expected the `func` on the same line as `ginkgo.It`, they weren't updated. Should the scripts perhaps enforce that they are on the same line? Tests that use two lines need to indent the function body one level deeper than tests that use one line, and it is inconsistent.

Reply:
I just checked, there are some places where sed is adding context parameters unnecessarily:
```
diff --git a/test/e2e/dra/dra.go b/test/e2e/dra/dra.go
index fc7529e682c..24aa9301249 100644
--- a/test/e2e/dra/dra.go
+++ b/test/e2e/dra/dra.go
@@ -61,7 +61,7 @@ var _ = ginkgo.Describe("[sig-node] DRA [Feature:DynamicResourceAllocation]", fu
                nodes := NewNodes(f, 1, 1)
                driver := NewDriver(f, nodes, networkResources) // All tests get their own driver instance.
                b := newBuilder(f, driver)
-               ginkgo.It("registers plugin", func() {
+               ginkgo.It("registers plugin", func(ctx context.Context) {
                        ginkgo.By("the driver is running")
                })
```

Here all the relevant work was done in `BeforeEach`, the test just exists to have something simple which exercises that and nothing else.

The other cases are `test/e2e/framework/log_test.go` and `test/e2e/framework/skipper/skipper_test.go` which are simpler when the callbacks don't use a context.

All of these could be converted, I just wanted to share my observation.

Reply:
Tim wanted to have this https://github.com/quasilyte/go-ruleguard that will allow us to write rules IIUIC :( ... but yeah, it has to be something more fancy that sed, it should understand the function signature

Reply:
I remember that one. I think we may be able to use it through golangci-lint because there is one plugin which accepts ruleguard configuration, but I haven't checked.

I would prefer to postpone enforcement of new rules until the dust has settled. Let's do that for 1.27.

Reply:
sure, it was a side comment, didn't mean to start fiddling with that now